// true for present, false for nothing
bool front
bool front_right
bool front_left
bool back